### PR TITLE
`externalContainerSelector` config implementation

### PR DESCRIPTION
**Pull request by JCCR**
This is an implementation based on what's proposed in https://github.com/hypothesis/client/pull/702

Please refer to the commit messages in this PR for the details on what's been done

### CODE REVIEW

**Comment by JCCR** [comment]
I'll soon post an example project that shows the usage of this..Just got the thought of adding this in the live reload page that appears in development

**Comment by JCCR** [comment]
Example project based on React with iframe'd content: https://github.com/JCCR/faux-reader

**Comment by judell** [comment]
This will be extremely helpful, thanks @JCCR.

**Comment by JCCR** [comment]
Rebased with latest master

**Comment by MohammadHader** [comment]
Please notice this issue #744

File: src/annotator/sidebar.coffee [hunk 1]
```diff
@@ -25,7 +25,13 @@ module.exports = class Sidebar extends Host
   gestureState: null
 
   constructor: (element, config) ->
+    if config.theme == 'clean' || config.externalContainerSelector
+      delete config.pluginClasses.BucketBar
```
Comment (robertknight) [inline src/annotator/sidebar.coffee]: I think it is surprising to have the constructor mutate the `config.pluginClasses` object here. I would have preferred to instead create a modified copy. There is also potentially a problem if anyone were ever to subclass `Sidebar` in future - although I would suggest that would almost certainly be a bad idea anyway.

**Pull request merged**